src/EnergyPlus/GroundHeatExchangers.cc

```diff
@@ -211,11 +469,762 @@ namespace GroundHeatExchangers {
+
+			// create empty json object
+			json json_in;
+
+			// read json_in data
+			try {
+				ifs >> json_in;
+				ifs.close();
+			} catch ( ... ) {
+				if ( !json_in.empty() ) {
+					// file exists, is not empty, but failed for some other reason
+					ShowWarningError( "eplusout.glhe contains invalid file format" );
+				}
+				ifs.close();
+				return;
+			}
+
+			for ( auto & existing_data : json_in ) {
+				if ( myCacheData["Phys Data"] == existing_data["Phys Data"] ) {
+					myCacheData["Response Factors"] = existing_data["Response Factors"];
+					gFunctionsExist = true;
+					break;
+				}
+			}
+
+			if ( gFunctionsExist ) {
+
+				// Setup the arrays
+				int numEntries = myCacheData["Response Factors"]["LNTTS"].size();
+
+				myRespFactors->time.dimension ( numEntries, 0.0 );
+				myRespFactors->LNTTS.dimension( numEntries, 0.0 );
+				myRespFactors->GFNC.dimension( numEntries, 0.0 );
+
+				// Populate the time array
+				int index = 1;
+				auto & j_time = myCacheData["Response Factors"]["time"];
+				for (json::iterator it = j_time.begin(); it != j_time.end(); ++it) {
+					myRespFactors->time( index ) = *it;
+					++index;
+				}
+
+				// Populate the lntts array
+				index = 1;
+				auto & j_lntts = myCacheData["Response Factors"]["LNTTS"];
+				for (json::iterator it = j_lntts.begin(); it != j_lntts.end(); ++it) {
+					myRespFactors->LNTTS( index ) = *it;
+					++index;
+				}
+
+				// Populate the g-function array
+				index = 1;
+				auto & j_gfnc = myCacheData["Response Factors"]["GFNC"];
+				for (json::iterator it = j_gfnc.begin(); it != j_gfnc.end(); ++it) {
+					myRespFactors->GFNC( index ) = *it;
+					++index;
+				}
+			}
+		}
+	}
+
+	//******************************************************************************
+
+	void
+	GLHEVert::writeCache()
+	{
+
+		// For convenience
+		using json = nlohmann::json;
+
+		if ( gio::file_exists( glheCacheFileName ) ) {
+			// file exists -- add data
+
+			// open file
+			std::ifstream ifs( glheCacheFileName );
+
+			// create empty json object
+			json json_in;
+
+			// read json_in data
+			try {
+				ifs >> json_in;
+				ifs.close();
+			} catch ( ... ) {
+				if ( !json_in.empty() ) {
+					// file exists, is not empty, but failed for some other reason
+					ShowWarningError( "Error reading from eplusout.glhe" );
+					ShowWarningError( "Data from previous eplusout.glhe not saved" );
+				}
+				ifs.close();
+			}
+
+			// empty json object for output writing
+			json json_out;
+
+			// add existing data to json_out
+			int i = 0;
+			for( auto & existing_data : json_in ) {
+				++i;
+				std::string case_name = "GHLE " + std::to_string( i );
+				json_out[case_name] = existing_data;
+			}
+
+			// add current data
+			std::string case_name = "GHLE " + std::to_string( i + 1 );
+			json_out[case_name] = myCacheData;
+
+			// open output file
+			std::ofstream ofs( glheCacheFileName );
+
+			// write data to file, set spacing at 2
+			ofs << std::setw( 2 ) << json_out;
```

Comment:
Each of these output operations should definitely be inside a try block and try to capture as many errors as possible.